+adi/+AD7173/Base.m

```diff
@@ -0,0 +1,155 @@
+classdef Base < adi.common.Rx & adi.common.RxTx & ...
+         matlabshared.libiio.base & adi.common.Attribute & ...
+         adi.common.RegisterReadWrite & adi.common.Channel
+    % AD7193 Precision ADC Class
+    % AD4111 is a 12 channel ADC with temperature/voltage/differential/current inputs
+
+    properties (Nontunable)
+        % SampleRate Sample Rate
+        %   Baseband sampling rate in Hz, specified as a scalar
+        %   in samples per second.
+        SampleRate = '19200'
+
+        % SamplesPerFrame Samples Per Frame
+        %   Number of samples per frame, specified as an even positive
+        %   integer.
+        SamplesPerFrame = 400
+    end
+
+    % isOutput
+    properties (Hidden, Nontunable, Access = protected)
+        isOutput = false
+    end
+
+    properties (Nontunable, Hidden, Constant)
+        Type = 'Rx'
+    end
+
+    properties (Nontunable, Hidden)
+        Timeout = Inf
+        kernelBuffersCount = 1
+        dataTypeStr = 'int32'
+        phyDevName
+        devName
+    end
+
+    properties (Hidden, Constant)
+        ComplexData = false
+    end
+
+    methods
+    %% Constructor
+        function obj = Base(phydev,dev,varargin)
+            coder.allowpcode('plain');
+            % Initialize the Rx object
+            obj = obj@matlabshared.libiio.base(varargin{:});
+            obj.enableExplicitPolling = false;
+            obj.EnabledChannels = 1;
+            obj.BufferTypeConversionEnable = true;
+            obj.phyDevName = phydev;
+            obj.devName = dev;
+            obj.uri = 'ip:analog.local';
+        end
+
+        function flush(obj)
+            % Flush the buffer
+            flushBuffers(obj);
+        end
+
+        function delete(obj)
+            % Destructor
+            delete@adi.common.RxTx(obj);
+        end
+
+        function set.SampleRate(obj, value)
+            % Set device sampling rate
+            obj.SampleRate = value;
+            if obj.ConnectedToDevice
+                obj.setDeviceAttributeRAW('sampling_frequency', value);
+            end
+        end
+
+        %% Check Voltage Scale
+        function rValue = get.VoltageScale(obj)
+            if obj.ConnectedToDevice
+                rValue = obj.getAttributeDouble('voltage0', 'scale', obj.isOutput);
```

Comment:
the default of 'voltage0' might not exists depending on how the dtb is written, how to account for that? I see `GetChannelAttrValue()` exists in `+Generic/Rx.m` but I'm unsure how to get the chnID from the first enabled channel?